clang/lib/Basic/Targets/X86.h

```diff
@@ -719,13 +719,33 @@ class LLVM_LIBRARY_VISIBILITY X86_64TargetInfo : public X86TargetInfo {
     Int64Type = IsX32 ? SignedLongLong : SignedLong;
     RegParmMax = 6;
 
+    bool isBIDEncoding = Opts.DFPEncoding == TargetOptions::BinaryEncoding::BID;
+    bool isDPDEncoding = Opts.DFPEncoding == TargetOptions::BinaryEncoding::DPD;
+
     // Pointers are 32-bit in x32.
-    resetDataLayout(IsX32 ? "e-m:e-p:32:32-p270:32:32-p271:32:32-p272:64:64-"
-                            "i64:64-f80:128-n8:16:32:64-S128"
-                          : IsWinCOFF ? "e-m:w-p270:32:32-p271:32:32-p272:64:"
-                                        "64-i64:64-f80:128-n8:16:32:64-S128"
-                                      : "e-m:e-p270:32:32-p271:32:32-p272:64:"
-                                        "64-i64:64-f80:128-n8:16:32:64-S128");
+    if (isBIDEncoding)
+      resetDataLayout(
+          IsX32       ? "e-d:-bid-m:e-p:32:32-p270:32:32-p271:32:32-p272:64:64-"
+                        "i64:64-f80:128-n8:16:32:64-S128"
+          : IsWinCOFF ? "e-d:bid-m:w-p270:32:32-p271:32:32-p272:64:"
+                        "64-i64:64-f80:128-n8:16:32:64-S128"
+                      : "e-d:bid-m:e-p270:32:32-p271:32:32-p272:64:"
+                        "64-i64:64-f80:128-n8:16:32:64-S128");
+    else if (isDPDEncoding)
+      resetDataLayout(
+          IsX32       ? "e-d:dpd-m:e-p:32:32-p270:32:32-p271:32:32-p272:64:64-"
+                        "i64:64-f80:128-n8:16:32:64-S128"
+          : IsWinCOFF ? "e-d:dpd-m:w-p270:32:32-p271:32:32-p272:64:"
+                        "64-i64:64-f80:128-n8:16:32:64-S128"
+                      : "e-d:dpd-m:e-p270:32:32-p271:32:32-p272:64:"
+                        "64-i64:64-f80:128-n8:16:32:64-S128");
+    else
+      resetDataLayout(IsX32 ? "e-m:e-p:32:32-p270:32:32-p271:32:32-p272:64:64-"
+                              "i64:64-f80:128-n8:16:32:64-S128"
+                      : IsWinCOFF ? "e-m:w-p270:32:32-p271:32:32-p272:64:"
+                                    "64-i64:64-f80:128-n8:16:32:64-S128"
+                                  : "e-m:e-p270:32:32-p271:32:32-p272:64:"
+                                    "64-i64:64-f80:128-n8:16:32:64-S128");
```

Comment:
Rather than duplicating the existing strings, I think this could be simplified with some string building. See the `WindowsX86_32TargetInfo` constructor later in this same file for an example.